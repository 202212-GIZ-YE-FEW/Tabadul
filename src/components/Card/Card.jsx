import React from "react";
import {
    CardDescription,
    CardImage,
    CardLink,
    Cardinfo,
    DateAndTime,
    IconsTitles,
    ItemIcon,
    LocationCategoryDiv,
    ProductCard,
    ProductDetails,
    ProductTitle,
} from "./Card.styled";
import { useRouter } from "next/router";

function Card(props) {
    const { productImage, title, id, location, category, date, description } =
        props;
    const router = useRouter();

    function checkLang(checkME) {
        if (router.locale === "en") {
            return checkME?.name;
        } else return checkME?.name_ar;
    }

    function getMonthName(itemDate) {
        const dateIntoArray = itemDate?.split("/");

        const date = new Date();
        date.setMonth(dateIntoArray?.[1] - 1);

        return `${dateIntoArray?.[0]} ${date.toLocaleString("en-US", {
            month: "long",
        })} ${dateIntoArray?.[2]}`;
    }
    return (
        <CardLink href={`/Products/${id}`} key={id} data-aos='zoom-in'>
            <ProductCard>
                <CardImage
                    src={productImage}
                    alt='CardImage'
                    width={258.4}
                    height={185.44}
                />
                <ProductDetails>
                    <ProductTitle>
                        {title?.length > 20
                            ? `${title?.substring(0, 18)} ...`
                            : title}
                    </ProductTitle>
                    <LocationCategoryDiv>
                        <Cardinfo>
                            <ItemIcon
                                src='/images/Vector.svg'
                                alt='Icon'
                                width={14}
                                height={16}
                            />
                            <IconsTitles>{checkLang(location)}</IconsTitles>
                        </Cardinfo>
                        <Cardinfo>
                            <ItemIcon
                                src='/images/tag-fill.svg'
                                alt='Icon'
                                width={14}
                                height={17}
                            />
                            <IconsTitles>{checkLang(category)}</IconsTitles>
                        </Cardinfo>
                    </LocationCategoryDiv>
                    <DateAndTime>{getMonthName(date)}</DateAndTime>
                    <CardDescription>
                        {description?.length > 50
                            ? `${description.substring(0, 50)} ...`
                            : description}
                    </CardDescription>
                </ProductDetails>
            </ProductCard>
        </CardLink>
    );
}

export default Card;
