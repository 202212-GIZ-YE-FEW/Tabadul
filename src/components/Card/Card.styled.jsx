import styled from "@emotion/styled";
import Image from "next/image";
import Link from "next/link";

export const CardLink = styled(Link)`
    text-decoration: none;
`;

export const ProductCard = styled.div`
    /* background-color: red; */
    width: 275px;
    height: 450px;
    background: #ffffff;
    box-shadow: 0px 0px 7px rgba(0, 0, 0, 0.25);
    border-radius: 20px;
    /* background: #ffffff; */
    margin: 10px;
    padding: 4px 4px 0 4px;

    &:hover {
        transform: translateY(-5px);
        box-shadow: 0 5px 15px rgba(0, 0, 0, 0.3);
    }
`;

export const CardImage = styled(Image)`
    object-fit: cover;
    object-position: center top;
    border-radius: 19px 19px 0px 0px;
    width: 100%;
    height: 244px;
`;

export const ProductDetails = styled.div`
    padding: 0 13px 0 13px;
    /* height: 100%; */
`;

export const ProductTitle = styled.p`
    font-weight: 600;
    font-size: 19px;
    color: #3c4347;
    margin: 0 0 0px 10px;
    word-wrap: break-word;
`;

export const LocationCategoryDiv = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: start;
    gap: 10px;
    padding: 9px 0px 2px 10px;
`;

export const Cardinfo = styled.div`
    display: flex;
    align-items: center;
`;
export const ItemIcon = styled(Image)`
    /* margin-left: 5px; */
`;

export const IconsTitles = styled.p`
    font-size: 16px;
    line-height: 30px;
    color: #3c4347;
    margin: 0 5px 0 7px;
`;
export const DateAndTime = styled.p`
    /* padding: 0 0 0 12px; */
    margin: 0 0 10px 12px;
    color: #8b8b8b;
    font-size: 13px;
`;

export const CardDescription = styled.p`
    border-top: 0.5px solid #d5dde1;
    padding-top: 18px;
    // font-family: "Poppins";
    font-size: 16px;
    font-weight: 400;
    width: 200px;
    color: #3c4347;
    margin-left: 10px;
    margin-top: 0;
    word-wrap: break-word;
`;
