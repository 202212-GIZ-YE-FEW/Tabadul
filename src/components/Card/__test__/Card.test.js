import renderer from "react-test-renderer";

import Card from "../Card";

jest.mock("next/router", () => ({
    useRouter() {
        return {
            pathname: "",
            // ... whatever else you you call on `router`
        };
    },
}));

it("renders correctly", () => {
    const tree = renderer.create(<Card />).toJSON();
    expect(tree).toMatchSnapshot();
});
