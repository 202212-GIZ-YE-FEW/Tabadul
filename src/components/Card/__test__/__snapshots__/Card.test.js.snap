// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<a
  className="css-1qj8w5r"
  data-aos="zoom-in"
  href="/Products/undefined"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onTouchStart={[Function]}
>
  <div
    className="css-j1sm4p"
  >
    <img
      alt="CardImage"
      className="css-tlhgah"
      data-nimg="1"
      decoding="async"
      height={185.44}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      src=""
      style={
        Object {
          "color": "transparent",
        }
      }
      width={258.4}
    />
    <div
      className="css-12mcu80"
    >
      <p
        className="css-1d0gjqw"
      />
      <div
        className="css-z8cq9n"
      >
        <div
          className="css-1povu0j"
        >
          <img
            alt="Icon"
            className="css-1vxmviq"
            data-nimg="1"
            decoding="async"
            height={16}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/images/Vector.svg"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={14}
          />
          <p
            className="css-1m4mhwh"
          />
        </div>
        <div
          className="css-1povu0j"
        >
          <img
            alt="Icon"
            className="css-1vxmviq"
            data-nimg="1"
            decoding="async"
            height={17}
            loading="lazy"
            onError={[Function]}
            onLoad={[Function]}
            src="/images/tag-fill.svg"
            style={
              Object {
                "color": "transparent",
              }
            }
            width={14}
          />
          <p
            className="css-1m4mhwh"
          />
        </div>
      </div>
      <p
        className="css-14yh97d"
      >
        undefined Invalid Date undefined
      </p>
      <p
        className="css-175v29n"
      />
    </div>
  </div>
</a>
`;
